src/cascadia/TerminalSettingsModel/ActionMap.h

```diff
@@ -66,8 +66,8 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         void AddAction(const Model::Command& cmd);
 
         // JSON
-        static com_ptr<ActionMap> FromJson(const Json::Value& json);
-        std::vector<SettingsLoadWarnings> LayerJson(const Json::Value& json, const bool withKeybindings = true);
+        static com_ptr<ActionMap> FromJson(const Json::Value& json, const OriginTag origin = OriginTag::None);
+        std::vector<SettingsLoadWarnings> LayerJson(const Json::Value& json, const OriginTag origin, const bool withKeybindings = true);

```

Comment:
Does the `origin` need to be an optional parameter? If it weren't we could ensure we never forget to pass the right tag.

Reply:
So the reason for this is in `SerializationTests.cpp` we call `FromJson` on several different classes (`Profile`, `ActionMap`, etc) in a template function, and not all have them take `OriginTag` as a parameter. We _should_ fix those tests to not behave like this and make `origin` a non-optional parameter, but I'm fine with leaving that as a follow up